### PR TITLE
Refactor Software Upgrade Documentation

### LINKED ISSUE #802: Update link in Software docs 0.29 that has steps for upgrading to 0.28
**Affected Document:**
https://docs.astronomer.io/software/upgrade-to-0-28

**Problem Type (Priority):**

- [ ] Error (Critical): This document has existing information that's incorrect and prevents readers from completing the documented setup (Add the `error/critical` label to this issue)
- [x] Misleading (Important): This document could benefit from additional context, guidance, or clarity (Add the `error/FYI` label to this issue)
- [ ] Typo (Low): Any issue related to spelling/ punctuation/ formatting (Add the `error/typo` label to this issue)

**Suggested Change (include links to relevant existing docs, support tickets, messages, or external content):**
Update the upgrade guide with info for upgrading to 0.29

**Additional Notes:**

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/787
Resolves #802 

If this looks good, I will also update the Upgrade docs in earlier doc versions so as to not cause confusion.

The riskiest part of this refactor is handling the 0.25 upgrade steps, which are a little more bespoke than the other versions.

### CODE REVIEW

File: software/upgrade-astronomer-stable.md [hunk 1]
```diff
@@ -1,62 +1,76 @@
...
 
-All stable and patch releases of Astronomer Software require a simple upgrade process. When an [LTS version](release-lifecycle-policy.md#release-channels) is released, additional upgrade guidance specific to that version will be made available.
-
 Follow this guide to upgrade to any stable or patch version of Astronomer Software. For information on new features and changes, refer to [Software Release Notes](release-notes.md).
 
 A few notes before you get started:
-- The upgrade process will not affect running Airflow tasks as long as `upgradeDeployments.enabled=false` is set in the script below.
-- Patch and stable version updates will not cause any downtime to Astronomer services, including the Software UI, the Astro CLI, and the Houston API.
+- The upgrade process will not affect running Airflow tasks as long as `upgradeDeployments.enabled=false` is set in your upgrade script.
+- Updates will not cause any downtime to Astronomer services, including the Software UI, the Astro CLI, and the Houston API.
```
Comment (kwcanuck) [inline software/upgrade-astronomer-stable.md:16]:
```suggestion
- Updates won’t cause Astronomer services downtime. This includes the Software UI, the Astro CLI, and the Houston API.
```

File: software/upgrade-astronomer-stable.md [hunk 2]
```diff
@@ -1,62 +1,76 @@
...
+## Step 1: Review Upgrade Considerations
 
-Astronomer v0.16.5 and beyond includes an improved upgrade process that allows Airflow Deployments to remain unaffected through a platform upgrade that includes changes to the [Astronomer Airflow Chart](https://github.com/astronomer/airflow-chart).
+The [Upgrade Considerations](upgrade-astronomer.md#upgrade-considerations) section of this document contains upgrade information for specific Astronomer versions. Review these notes before starting your own upgrade process.  
 
-Now, Airflow Chart changes only take effect when another restart event is triggered by a user. This includes a code push, a change to Environment Variables, or an adjustment to Deployment resources.
+## Step 2: Check Permissions
 
-:::
+Version upgrades can be initiated only by a user with System Admin permissions. To confirm you're an Astronomer System Admin, check that you have access to the **System Admin** menu in the Software UI:
```
Comment (kwcanuck) [inline software/upgrade-astronomer-stable.md:24]:
```suggestion
You need Astronomer System Admin permissions to complete version upgrades. To confirm you're an Astronomer System Admin, check that you have access to the **System Admin** menu in the Software UI:
```

File: software/upgrade-astronomer-stable.md [hunk 3]
```diff
@@ -1,62 +1,76 @@
...
+kubectl auth can-i create jobs --namespace <your-astronomer-namespace>
+```
 
-:::
+If all commands return `yes`, then you have the appropriate Kubernetes permissions.
+
+## Step 3: Back Up Your Database
 
-## Step 1: Ensure You Have a Copy of Your Astronomer `config.yaml` File
+Before you perform an upgrade, back up your Astronomer database by following recommendations from your cloud provider or making a backup request to your organization's database administrator.
```
Comment (kwcanuck) [inline software/upgrade-astronomer-stable.md:40]:
```suggestion
Before you perform an upgrade, back up your Astronomer database. To create a backup, follow the instructions provided by your cloud provider, or ask your organization's database administrator for assistance.
```

**Review by cmarteepants — APPROVED** [review]
Looks good! Can be done as a future change, but we should publish an upgrade path similar to Gitlab: https://docs.gitlab.com/ee/update/#upgrade-paths

(I generally think their docs are well structured)

File: software/upgrade-astronomer-stable.md [hunk 4]
```diff
@@ -96,25 +110,138 @@ If you do not specify a patch version above, the script will automatically pull
...
+
+    For example, all of the following images would work for this upgrade:
+
+    - `quay.io/astronomer/ap-airflow:1.10.15-7-buster`
+    - `quay.io/astronomer/ap-airflow:2.0.0-3-buster-onbuild`
+    - `quay.io/astronomer/ap-airflow:2.0.2-buster-onbuild`
+    - `quay.io/astronomer/ap-airflow:2.2.2-onbuild`
+
+- **Helm**: Your version must be 3.6 ≤ 3.8.
+
```
[the remark this answers is not in the document]
Comment (jwitz) [reply software/upgrade-astronomer-stable.md:182]: If there's any more compatibility to know about maybe, but I think we cover everything for the upgrade here. Not worth people second guessing whether there's another compatibility concern to worry about.

File: software/upgrade-astronomer-stable.md [hunk 5]
```diff
@@ -96,25 +110,138 @@ If you do not specify a patch version above, the script will automatically pull
...
 This topic contains information about upgrading to specific versions of Astronomer Software. This includes notes on breaking changes, database migrations, and other considerations that might depend on your use case.
 
-### Upgrading from Earlier Minor Versions to 0.29
+### Upgrading to 0.29
+
+#### Minimum Pre-Upgrade Version
+
+If you are currently on a pre-0.28 version of Software, you must upgrade to Software 0.28 before upgrading to 0.29. A direct upgrade from a pre-0.28 version to 0.29 is not possible.
+
+Follow the standard installation guide to upgrade to Software 0.28, then repeat the process to upgrade to 0.29.
```
[the remark this answers is not in the document]
Comment (jwitz) [reply software/upgrade-astronomer-stable.md:147]: To this doc? I think we're okay here

**Comment by jwitz** [comment]
Tested redirects, double-checked file structure in preview site- we should be good to go!

**Pull request merged**